src/main/java/org/jabref/cli/ArgumentProcessor.java

```diff
@@ -535,7 +535,7 @@ private void regenerateCitationKeys(List<ParserResult> loaded) {
         String engine = split[0];
         String query = split[1];
 
-        Set<SearchBasedFetcher> fetchers = WebFetchers.getSearchBasedFetchers(Globals.prefs.getImportFormatPreferences());
+        Set<SearchBasedFetcher> fetchers = WebFetchers.getSearchBasedFetchers(Globals.prefs);
```

Comment:
@calixtus We should discuss this - on our last devcall, we said, we wanted to keep the preferences objects small. Can't we keep the `ImportFormatPreferences` here?

Reply:
I was working on this PR yesterday. The problem is, the fetchers are being constructed with the ImportFormatPreferences when combobox with possible fetchers in the sidepane is constructed. If you change the api key afterwards, ImportFormatPreferences is already created and not altered anymore. in that case you would have to restart JabRef everytime you enter a custom api key, if you made a mistake in entering etc. Otherwise by using the preferences service a new api key object is being created on time when needed.
I know, the scope of the preferences service becomes larger, but i wouldn't know any better solution right now.

Reply:
Timely updates will bring a better user experience, but this is not a frequently used feature, maybe we have to make some compromises.